### DISCUSSION TITLE
Renovate full Docker image uses php 7.4

### DISCUSSION THREAD

**Discussion by kolaente** [renovatebot/renovate#17104] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.153.4

### Please select which platform you are using if self-hosting.

Gitea

### If you're self-hosting Renovate, tell us what version of the platform you run.

Gitea 1.17.0

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

PHP projects which require at least php 8.1 fail to update with this error message posted by the renovate bot:

```
Command failed: composer update fakerphp/faker --with-dependencies --ignore-platform-req='ext-*' --ignore-platform-req='lib-*' --no-ansi --no-interaction --no-scripts --no-autoloader --no-plugins
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires php 8.1.* but your php version (7.4.30) does not satisfy that requirement.
```

This could be solved by either adding the ` --ignore-platform-req=php` to the composer upgrade command or upgrading the 'used php version in the docker image. The first one is probably easier and backwards compatible.

### Relevant debug logs

IMHO everything relevant is in the description, please tell me if you need any other specific logs.

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Not a bug. The full Renovate image is built using this Dockerfile: https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile

You may download it and customize if you need, or use the slim Docker image with binarySource=docker

**Reply by kolaente** [reply]
Given php 7.4 is already in the "security fixes only" phase and will be EOL by November I don't think it's very user-friendly to expect people to build their own image. This likely affects users of the hosted version as well.

Where would I add the `--ignore-platform-req=php` parameter? Would you accept a PR for that?

**Reply by viceice** [reply]
nope, the hosted app uses docker mode, so it will use a sidecar container with the right php version

https://docs.renovatebot.com/self-hosted-configuration/#binarysource

**Reply by kolaente** [reply]
So when I change that parameter to `docker` it should work?

**Reply by viceice** [reply]
it depends how are you runnig renovate

we've some more docker samples here[^1]

[^1]: https://github.com/renovatebot/docker-renovate#usage
